ovgs.proto

```diff
@@ -172,45 +171,69 @@ message GetOwnershipVoucherResponse {
-  rpc AssignUserRole(AssignUserRoleRequest) returns (AssignUserRoleResponse);
-  // UnassignUserRole removes roles of a user in a named group.
-  rpc UnassignUserRole(AssignUserRoleRequest) returns (AssignUserRoleResponse);
+  // CreateUserRole assigns a role to a user in a named group.
+  // Errors will be returned:
+  // ALREADY_EXISTS if user already exists in the group
+  // PERMISSION_DENIED if the user doesn't have access to the group
+  rpc CreateUserRole(AssignUserRoleRequest) returns (AssignUserRoleResponse);
+  // DeleteUserRole removes roles of a user in a named group.
+  // Errors will be returned:
+  // NOT_FOUND if the group doesn't exist
+  // PERMISSION_DENIED if user doesn't have access to the group
+  rpc DeleteUserRole(AssignUserRoleRequest) returns (AssignUserRoleResponse);
   // GetUserRole returns the roles the user is assigned.
+  // Errors will be returned:
+  // NOT_FOUND if the group doesn't exist
+  // PERMISSION_DENIED if the user doesn't have access to the group
   rpc GetUserRole(GetUserRoleRequest) returns (GetUserRoleResponse);
 
   // CreateSerial assigns the serial to that group.
+  // Errors will be returned:
+  // ALREADY_EXISTS if serial_number already exists in the group
+  // PERMISSION_DENIED if the user doesn't have access to the group
   rpc CreateSerial(CreateSerialRequest) returns (CreateSerialResponse);
-  // UnassignSerial removes the serial from the group.
+  // DeleteSerial removes the serial from the group.
+  // Errors will be returned:
+  // NOT_FOUND if the group doesn't exist
+  // PERMISSION_DENIED if user doesn't have access to the group
   rpc DeleteSerial(DeleteSerialRequest) returns (DeleteSerialResponse);
   // GetSerial returns all the facts about the serial number.
+  // Errors will be returned:
+  // NOT_FOUND if the group doesn't exist
+  // PERMISSION_DENIED if the user doesn't have access to the group
   rpc GetSerial(GetSerialRequest) returns (GetSerialResponse);
 
-  // UpdateDomainCert sets the values for the certificate.
-  rpc UpdateDomainCert(setDomainCertRequest) returns (setDomainCertResponse);
-  // readDomainCert reveals the details of the certificate.
-  rpc GetDomainCert(readDomainCertRequest) returns (readDomainCertResponse);
-  // deleteDomainCert deletes the cert from the database.
-  rpc DeleteDomainCert(deleteDomainCertRequest) returns (deleteDomainCertResponse);
+  // CreateDomainCert creates the certificate in the group.
+  // Errors will be returned:
+  // ALREADY_EXISTS if cert already exists in the group
+  // PERMISSION_DENIED if the user doesn't have access to the group
+  rpc CreateDomainCert(CreateDomainCertRequest) returns (CreateDomainCertResponse);
+  // DeleteDomainCert deletes the cert from the database.
+  // Errors will be returned:
+  // NOT_FOUND if the cert doesn't exist
+  // PERMISSION_DENIED if user doesn't have access to the group
+  rpc DeleteDomainCert(DeleteDomainCertRequest) returns (DeleteDomainCertResponse);
```

Comment:
picking just 1 of the create/delete/get sets here, is the plan for 'oops, change domaincert comment (or other attribute)' really: "delete, then re-Create" (there is no update).

That might be fine, or we may not expect to do updates.
There may be gotchas if the delete cascades (for instance) or the caller would have to gather all of the state first, before delete, then re-home all the  parts of the tree/connetions.